be/src/storage/compaction_manager.cpp

```diff
@@ -385,6 +385,125 @@ void CompactionManager::clear_tasks() {
     _cumulative_compaction_concurrency = 0;
 }
 
+// for http action
+void CompactionManager::get_running_status(std::string* json_result) {
+    int32_t max_task_num;
+    int64_t base_task_num;
+    int64_t cumulative_task_num;
+    int64_t running_task_num;
+    int64_t candidate_num;
+    std::unordered_map<std::string, uint16_t> data_dir_to_base_task_num;
+    std::unordered_map<std::string, uint16_t> data_dir_to_cumulative_task_num;
+    vector<int64_t> running_tablet_ids;
+    vector<int64_t> candidate_ids;
+    {
+        std::lock_guard lg(_tasks_mutex);
+        max_task_num = _max_task_num;
+        base_task_num = _base_compaction_concurrency;
+        cumulative_task_num = _cumulative_compaction_concurrency;
+        running_task_num = _running_tasks.size();
+        running_tablet_ids.reserve(running_task_num);
+        for (auto it : _running_tasks) {
+            running_tablet_ids.push_back(it->tablet()->tablet_id());
+        }
+        for (auto it : _data_dir_to_base_task_num_map) {
+            data_dir_to_base_task_num[it.first->path()] = it.second;
+        }
+        for (auto it : _data_dir_to_cumulative_task_num_map) {
+            data_dir_to_cumulative_task_num[it.first->path()] = it.second;
+        }
+        std::lock_guard lg_candidates(_candidates_mutex);
+        candidate_num = _compaction_candidates.size();
+        candidate_ids.reserve(candidate_num);
+        for (const auto& it : _compaction_candidates) {
+            candidate_ids.push_back(it.tablet->tablet_id());
+        }
+    }
+
+    rapidjson::Document root;
+    root.SetObject();
+
+    rapidjson::Value max_task_num_value;
+    max_task_num_value.SetInt(max_task_num);
+    root.AddMember("max_task_num", max_task_num_value, root.GetAllocator());
+
+    rapidjson::Value base_task_num_value;
+    base_task_num_value.SetInt64(base_task_num);
+    root.AddMember("base_task_num", base_task_num_value, root.GetAllocator());
+
+    rapidjson::Value cumulative_task_num_value;
```

Comment:
We have used simdjson to replace the rapidjson for performance. The old rapidjson will be removed gruadually.